source/set-return-type.d.ts

```diff
@@ -0,0 +1,29 @@
+type IsAny<T> = 0 extends (1 & T) ? true : false; // https://stackoverflow.com/a/49928360/3406963
+type IsNever<T> = [T] extends [never] ? true : false;
+type IsUnknown<T> = IsNever<T> extends false ? T extends unknown ? unknown extends T ? IsAny<T> extends false ? true : false : false : false : false;
+
+/**
+Get a function type based on another, changing the return type to one of your choice.
+
+Use-case: You want to define a wrapped function that returns something different while receiving the same parameters.
+
+@example
+```
+import {SetReturnType} from 'type-fest';
+
+type MyFunction = (foo: SomeType, bar: unknown) => SomeOtherType;
+
+type MyWrappedFunction = SetReturnType<MyFunction, boolean>;
+// type MyWrappedFunction = (foo: SomeType, bar: unknown) => boolean;
+```
+*/
+export type SetReturnType<Fn extends (...args: any[]) => any, TypeToReturn> =
+	// Just using `Parameters<Fn>` isn't ideal because it doesn't handle the `this` fake parameter
```

Comment:
Is this intended TS behavior or a bug? If it's a bug, it would be good to open an issue on TS.

Reply:
That is a good question, I don't know the answer :sweat_smile:

I have never really used the `this` fake parameter explicitly... I recalled its existence while writing the PR, tested for it and noticed `Parameters<Fn>` doesn't catch it... Didn't even think it might be a bug :sweat_smile:

Reply:
Done! https://github.com/microsoft/TypeScript/issues/41072 and https://github.com/microsoft/TypeScript/issues/41073